From 875e545368dbac6acb79a72dbd39316c71de3fa6 Mon Sep 17 00:00:00 2001
From: Arpit Agarwal <arp@apache.org>
Date: Thu, 11 Feb 2016 20:40:03 -0800
Subject: [PATCH 1942/2848] HDFS-9790. HDFS Balancer should exit with a proper
 message if upgrade is not finalized (Contributed
 by Xiaobing Zhou)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java

(cherry picked from commit 9b959f3ee70cfdaf532d72f0bbbd1a4be0be12aa)

Change-Id: Ib8285d92e58436246ea78ee6c968c65e567a1f2e
---
 .../hadoop/hdfs/server/balancer/Balancer.java      |    2 ++
 1 file changed, 2 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
index 1b880c2..4a682e1 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/balancer/Balancer.java
@@ -579,6 +579,8 @@ Result runOneIteration() {
       // Should not run the balancer during an unfinalized upgrade, since moved
       // blocks are not deleted on the source datanode.
       if (!runDuringUpgrade && nnc.isUpgrading()) {
+        System.err.println("Balancer exiting as upgrade is not finalized, "
+            + "please finalize the HDFS upgrade before running the balancer.");
         return newResult(ExitStatus.UNFINALIZED_UPGRADE, bytesLeftToMove, -1);
       }
 
-- 
1.7.9.5

